pkg/controllers/management/cluster/clustercontroller.go

```diff
@@ -0,0 +1,121 @@
+package cluster
+
+import (
+	"github.com/rancher/rke/cloudprovider/aws"
+	"github.com/rancher/rke/cloudprovider/azure"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type controller struct {
+	clusterClient v3.ClusterInterface
+	clusterLister v3.ClusterLister
+	nodeLister    v3.NodeLister
+}
+
+func Register(management *config.ManagementContext) {
+	c := controller{
+		clusterClient: management.Management.Clusters(""),
+		clusterLister: management.Management.Clusters("").Controller().Lister(),
+		nodeLister:    management.Management.Nodes("").Controller().Lister(),
+	}
+
+	c.clusterClient.AddHandler("clusterCreateUpdate", c.clusterSync)
+}
+
+func (c *controller) clusterSync(key string, obj *v3.Cluster) error {
+	var updated bool
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return nil
+	}
+
+	cluster, err := c.clusterClient.Get(key, metav1.GetOptions{})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	if k8serrors.IsNotFound(err) {
+		return nil
+	}
+
+	if cluster.Status.Driver == v3.ClusterDriverImported {
+		return nil
+	}
+	capabilities := v3.Capabilities{}
+	capabilities.NodePortRange = "30000-32767"
+
+	if cluster.Spec.GoogleKubernetesEngineConfig != nil {
+		capabilities.L4LoadBalancer = c.L4Capability(true, "GCLB", []string{"TCP", "UDP"}, true)
```

Comment:
lets think about making constants for the strings that are repeated several times